app/src/main/java/de/westnordost/streetcomplete/quests/internet_access/AddInternetAccess.java

```diff
@@ -0,0 +1,52 @@
+package de.westnordost.streetcomplete.quests.internet_access;
+
+
+import android.os.Bundle;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.osm.SimpleOverpassQuestType;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+
+public class AddInternetAccess extends SimpleOverpassQuestType
+{
+	@Inject public AddInternetAccess(OverpassMapDataDao overpassServer) { super(overpassServer); }
+
+	@Override
+	protected String getTagFilters() {
+		return "nodes, ways, relations with " +
+				" amenity ~ restaurant|cafe|ice_cream|fast_food|bar|pub|biergarten|food_court|cinema|library" +
```

Comment:
I think the quest should be limited to a much smaller list. Thinking worldwide here, of the amenities, only library should stay

Reply:
Actually all of these may not have WiFi at all in some countries. That includes libraries, which - if they do - only provide LAN.

I have however no real idea, where one could get statistics from, other, than maybe existing OSM stats.

Reply:
But there are less (hotels,...) and the information has a higher value as for restaurants etc (because you spend more time there), especially for OSM surveyors actually. Also, hotels etc are usually more long-lived than those amenities, so the information is also less volatile.